### PR TITLE
No diff clean up

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes Nada
 - Clean up in prep for follow up UnitarySystem cleaning

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by rraustad** [comment]
I'm not sure where this came from but it's a show stopper if true.

image

File: src/EnergyPlus/UnitarySystem.cc [hunk 1]
```diff
@@ -4315,17 +4248,17 @@ namespace UnitarySystems {
         if (UtilityRoutines::SameString(loc_supFanPlace, "BlowThrough")) this->m_FanPlace = FanPlace::BlowThru;
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:4248]: Is there no `enum class` defined for this yet?

File: src/EnergyPlus/UnitarySystem.cc [hunk 2]
```diff
@@ -4315,17 +4248,17 @@ namespace UnitarySystems {
...
         if (UtilityRoutines::SameString(loc_supFanPlace, "DrawThrough")) this->m_FanPlace = FanPlace::DrawThru;
         if (this->m_FanPlace == FanPlace::NotYetSet && this->m_FanExists) {
-            ShowSevereError(state, cCurrentModuleObject + " = " + thisObjectName);
-            ShowContinueError(state, "Illegal Fan Placement = " + loc_supFanPlace);
+            ShowSevereError(state, format("{} = {}", cCurrentModuleObject, thisObjectName));
+            ShowContinueError(state, format("Illegal Fan Placement = {}", loc_supFanPlace));
             errorsFound = true;
         }
 
         std::string loc_supFanOpMode = input_data.supply_air_fan_operating_mode_schedule_name;
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:4256]: I know you didn't write this line, but `loc_supFanOpMode` is a deep copy of the string in `input_data`. Do you really need a copy here? Or is a reference (either `std::string const &` or `std::string_view`) sufficient? I would be the latter.

File: src/EnergyPlus/UnitarySystem.cc [hunk 3]
```diff
@@ -125,49 +125,17 @@ namespace UnitarySystems {
...
-                                 int const ZoneOAUnitNum,
-                                 Real64 const OAUCoilOutTemp,
-                                 bool const ZoneEquipment,
-                                 Real64 &sysOutputProvided,
-                                 Real64 &latOutputProvided)
     {
         DataHVACGlobals::CompressorOperation CompressorOn = DataHVACGlobals::CompressorOperation::Off;
 
         // Obtains and Allocates unitary system related parameters from input file
         if (this->m_ThisSysInputShouldBeGotten) {
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:132]: I'm not going to harp on this too much because it's not super important, but ... `this->` and `m_` are redundant. 

`m_` is "scoping Hungarian notation" to indicate that a variable has object member scope, i.e., it's a member of the object on which the current method is being called, i.e., a member of `this`, i.e., equivalent `this->m_`. The two other scopes are local (declared and live within the function, this includes parameters) and global. In "scoping Hungarian", global variables begin with `g_` and local variables don't have a prefix at all.

The C/C++ scoping precedence rule is local > member > global, i.e., the compiler will first look for a variable among the list of local variables, if there is no match there it will look for it in the list of member variables (and if it finds it there it will implicitly add the `this->`), finally it will look in the list of global variables. The only time `this->` is really necessary is if there is a local variable and a member variable of the same name and you want to refer to the member variable. However, if all member variables start with `m_` and all local variables do not then this situation never comes up. 

TL;DR If you are already using `m_` for member variables (which I think is good practice), you don't need `this->`. Using `this->` doesn't make the code slower, just more verbose and IMO difficult to read.

File: src/EnergyPlus/UnitarySystem.cc [hunk 4]
```diff
@@ -5992,7 +5925,7 @@ namespace UnitarySystems {
             if (UtilityRoutines::SameString(loc_m_ISHundredPercentDOASDXCoil, "Yes")) {
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:5925]: There is a `getYesNo` similar to `getEnumerationValue`.

File: src/EnergyPlus/UnitarySystem.cc [hunk 5]
```diff
@@ -3858,7 +3789,7 @@ namespace UnitarySystems {
             this->ControlZoneNum = UtilityRoutines::FindItemInList(loc_controlZoneName, state.dataHeatBal->Zone);
         } else if (this->m_ControlType == UnitarySysCtrlType::Load || this->m_ControlType == UnitarySysCtrlType::CCMASHRAE) {
             if (this->m_sysType < SysType::PackagedAC) {
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:3791]: Doing `<`, `>` comparisons on enumerations is not good practice. Better (from both a robustness standpoint and a readability standpoint) to do a bunch of `==` comparisons.

File: src/EnergyPlus/UnitarySystem.cc [hunk 6]
```diff
@@ -4235,19 +4168,19 @@ namespace UnitarySystems {
...
                 } else {
                     isNotOK = false;
                     ValidateComponent(state, loc_fanType, loc_m_FanName, isNotOK, cCurrentModuleObject);
                     if (isNotOK) {
-                        ShowContinueError(state, "Occurs in " + cCurrentModuleObject + " = " + thisObjectName);
+                        ShowContinueError(state, format("Occurs in {} = {}", cCurrentModuleObject, thisObjectName));
                         errorsFound = true;
                     } else { // mine data from fan object
                         // Get the fan index
                         Fans::GetFanIndex(state, loc_m_FanName, this->m_FanIndex, errFlag);
```
Comment (amirroth) [inline src/EnergyPlus/UnitarySystem.cc:4181]: I'm sure you already know this (at least implicitly), but once you've gotten the FanIndex here you don't need to keep searching for it again and again in `GetFanDesignVolumeFlowRate`, `GetFanInletNode`, `GetFanOutletNode`, etc. You can just use declare `auto &fan = state.dataFans->Fans(FanIndex);` and just pick off the fields you need from that.

File: src/EnergyPlus/UnitarySystem.cc (hunk 4 above)
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc:5925]: Not required field. Blank input is allowed so getYesNo won't work here.

File: src/EnergyPlus/UnitarySystem.cc (hunk 5 above)
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc:3791]: I cleaned up all of these.

File: src/EnergyPlus/UnitarySystem.cc (hunk 6 above)
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc:4181]: Lot's of this completed, more to go.

File: src/EnergyPlus/UnitarySystem.cc (hunk 2 above)
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc:4256]: I did write it and these loc_* were temporary and I began removing them in this branch.

File: src/EnergyPlus/UnitarySystem.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc:4248]: I'll get to this eventually.

**Comment by rraustad** [comment]
This branch is ready for review and merge. I don't intent to do more work on this branch.

File: src/EnergyPlus/UnitarySystem.cc [hunk 7]
```diff
@@ -125,49 +125,17 @@ namespace UnitarySystems {
...
-                                 bool const FirstHVACIteration,
-                                 int const AirLoopNum,
-                                 int &CompIndex,
-                                 bool &HeatActive,
-                                 bool &CoolActive,
-                                 int const ZoneOAUnitNum,
-                                 Real64 const OAUCoilOutTemp,
-                                 bool const ZoneEquipment,
-                                 Real64 &sysOutputProvided,
-                                 Real64 &latOutputProvided)
```
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:154]: Clean up started here and then included a lot of recent suggestions for improvement. This function was added when sysOutputProvided and latOutputProvided were optional arguments and is no longer needed.

File: src/EnergyPlus/UnitarySystem.cc [hunk 8]
```diff
@@ -125,49 +125,17 @@ namespace UnitarySystems {
...
     {
         DataHVACGlobals::CompressorOperation CompressorOn = DataHVACGlobals::CompressorOperation::Off;
 
         // Obtains and Allocates unitary system related parameters from input file
         if (this->m_ThisSysInputShouldBeGotten) {
-            // Get the unitary system input
```
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:160]: General code clean up from here to end.

File: src/EnergyPlus/UnitarySystem.cc (hunk 6 above)
Comment (rraustad) [reply src/EnergyPlus/UnitarySystem.cc:4181]: Most of the indexing is completed. I wouldn't doubt if there is a few more places for this type of change.

File: src/EnergyPlus/UnitarySystem.cc (hunk 4 above)
Comment (amirroth) [reply src/EnergyPlus/UnitarySystem.cc:5925]: You just have to test for blank first. Not a big deal for now, just something to know for later.

**Comment by Myoldmopar** [comment]
@rraustad thanks for this nice no-diff cleanup.  I realize any remaining issues could be addressed on the actual Unitary cleanup, so I'll just take this as it is.  CI is all happy (GPU diffs on Linux).  I will pull in develop since it just moved and make sure it's happy, but assuming it is, I think this will merge in a bit.

**Comment by Myoldmopar** [comment]
Running fine, merging.
